core/src/main/java/org/elasticsearch/action/main/MainResponse.java

```diff
@@ -137,4 +138,26 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
     public static MainResponse fromXContent(XContentParser parser) {
         return PARSER.apply(parser, null);
     }
+
+    @Override
```

Comment:
can you get this in separately and backport it? I am fine with pushing without a PR if that's handy. I worry that this PR is targeted to master only and depends on stuff that's outside of the rest-high-level folder.